onebusaway-android/src/main/java/org/onebusaway/android/ui/TripResultsFragment.java

```diff
@@ -0,0 +1,296 @@
+        return (List<Itinerary>) getArguments().getSerializable(OTPConstants.ITINERARIES);
+    }
+
+    private CustomAddress getFromAddress() {
+        return new TripRequestBuilder(getArguments()).getFrom();
+    }
+
+    private CustomAddress getToAddress() {
+        return new TripRequestBuilder(getArguments()).getTo();
+    }
+
+    private Date getRequestDate() {
+        return new TripRequestBuilder(getArguments()).getDateTime();
+    }
+
+    private class RoutingOptionPicker {
+        LinearLayout linearLayout;
+        TextView titleView;
+        TextView durationView;
+
+        Itinerary itinerary;
+        int rank;
+
+        RoutingOptionPicker(View view, int linearLayout, int titleView, int durationView) {
+            this.linearLayout = (LinearLayout) view.findViewById(linearLayout);
+            this.titleView = (TextView) view.findViewById(titleView);
+            this.durationView = (TextView) view.findViewById(durationView);
+
+            this.linearLayout.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    updateInfo();
+                    updateMap();
+                    RoutingOptionPicker.this.select();
+                }
+            });
+        }
+
+        void select() {
+            for (RoutingOptionPicker picker : mOptions)
+                picker.linearLayout.setBackgroundColor(getResources().getColor(R.color.trip_option_background));
+            linearLayout.setBackgroundResource(R.drawable.bottom_line_grey_blue);
+        }
+
+
+        void setItinerary(int rank) {
+            this.rank = rank;
+
+            List<Itinerary> trips = getItineraries();
+            if (rank >= trips.size()) {
+                this.itinerary = null;
+                linearLayout.setVisibility(View.GONE);
+                return;
+            }
+
+            this.itinerary = trips.get(rank);
+            String title = new DirectionsGenerator(itinerary.legs, getContext()).getItineraryTitle();
+            String duration = formatTimeString(itinerary.startTime, itinerary.duration * 1000);
+            titleView.setText(title);
+            durationView.setText(duration);
+        }
+
+        void updateInfo() {
+            DirectionsGenerator gen = new DirectionsGenerator(itinerary.legs, getActivity().getApplicationContext());
+            List<Direction> directions = gen.getDirections();
+            Direction direction_data[] = directions.toArray(new Direction[directions.size()]);
+
+            DirectionExpandableListAdapter adapter = new DirectionExpandableListAdapter(
+                    getActivity(),
+                    R.layout.list_direction_item, R.layout.list_subdirection_item, direction_data);
+
+            mDirectionsListView.setAdapter(adapter);
+
+            mDirectionsListView.setGroupIndicator(null);
+
+            mRealtimeService.onItinerarySelected(itinerary, rank);
```

Comment:
We may want to move this into `onPause()`, so the notification only start when the user leaves the view of the trip planning results.  In some cases, I'm currently getting notified each time I tap on the different tabs for each trip plan result, which is confusing.